src/core/border_router/routing_manager.cpp

```diff
@@ -3601,6 +3600,11 @@ void RoutingManager::PdPrefixManager::EvaluateStateChange(Dhcp6PdState aOldState
     case kDhcp6PdStateRunning:
         break;
     }
+#if OPENTHREAD_CONFIG_BORDER_ROUTING_DHCP6_PD_ENABLE
```

Comment:
I think the whole `PdPrefixManager` class and all its methods are under a  `#if OPENTHREAD_CONFIG_BORDER_ROUTING_DHCP6_PD_ENABLE` block.

So dont need to add this `#if` check again here?